packages/react/src/ReferenceInput/ReferenceInput.test.tsx

```diff
@@ -178,4 +179,60 @@ describe('ReferenceInput', () => {
     expect(screen.getByPlaceholderText('Resource Type')).toBeInTheDocument();
     expect(screen.queryByTestId('reference-input-resource-type-select')).not.toBeInTheDocument();
   });
+
+  test.only('Handle profile target type', async () => {
+    const FishPatientProfileSD = FishPatientResources.getFishPatientProfileSD();
+    const blinky = FishPatientResources.getBlinkyTheFish();
+    await medplum.createResource(blinky);
+
+    indexStructureDefinitionBundle([FishPatientProfileSD], FishPatientProfileSD.url);
+    setup({
+      name: 'foo',
+      targetTypes: [FishPatientProfileSD.url, 'Patient'],
+      placeholder: 'My placeholder',
+    });
+
+    // Before the profile is fetch, the URL is shown
+    expect(screen.getByDisplayValue(FishPatientProfileSD.url)).toBeInTheDocument();
+
+    // wait for the profile to be fetched
+    await waitFor(() => screen.getByText('Fish Patient'));
+
+    // After the profile is fetched, the URl is replaced by the title
+    expect(screen.queryByDisplayValue(FishPatientProfileSD.url)).toBeNull();
+    expect(screen.getByDisplayValue('Fish Patient')).toBeInTheDocument();
+
+    // Enter "B"
+    await act(async () => {
+      fireEvent.change(screen.getByPlaceholderText('My placeholder'), { target: { value: 'B' } });
+    });
+
+    // Wait for the autocomplete timeout
+    await act(async () => {
+      jest.advanceTimersByTime(1000);
+    });
+
+    // Blinky is a fish, Bart is not
+    expect(screen.queryByText('Blinky')).toBeInTheDocument();
+    expect(screen.queryByText('Bart Simpson')).toBeNull();
+
+    // Select "Patient" resource type
+    await act(async () => {
+      fireEvent.change(screen.getByTestId('reference-input-resource-type-select'), { target: { value: 'Patient' } });
+    });
+
+    // Refocus the input; "B" still as the last value
+    await act(async () => {
+      fireEvent.focus(screen.getByPlaceholderText('My placeholder'));
+    });
+
+    // Wait for the autocomplete timeout
+    await act(async () => {
+      jest.advanceTimersByTime(1000);
```

Comment:
Interesting, do you have to advance timers in an act? That actually might explain some problems I had before

Reply:
It appears so! I had blindly copy/pasted this from elsewhere in the codebase. I just tried removing the `await act` wrapper and the test failed.

Reply:
@ThatOneBro - You should only need to call `jest.advanceTimers` when using [Jest Fake Timers](https://jestjs.io/docs/timer-mocks)

Aside: I have a long standing mental todo to go back and review our use of Jest fake timers.  I'm pretty sure there are some files that should use them, but currently do not, and therefore actually `sleep(1000)`, which adds up.  `packages/core/src/client.test.ts` in paticular